app/models/requests/DataRequest.scala

```diff
@@ -26,4 +26,13 @@ case class DataRequest[A](request: UserRequest[A],
 
   val internalId: String = request.internalId
   val ern: String = request.ern
+
+  val userTypeFromErn: UserType = ern.take(4).toUpperCase match {
+    case "GBRC" => GreatBritainRegisteredConsignor
+    case "XIRC" => NorthernIrelandRegisteredConsignor
+    case "GBWK" => GreatBritainWarehouseKeeper
+    case "XIWK" => NorthernIrelandWarehouseKeeper
+    case _ => Unknown
```

Comment:
I think there's more types than this. Might be worth checking with Mitchell, Casey or Tom for a complete list.

Reply:
I'd argue that this is also potentially scope creep as the stories with this and the destination type should've been played before this one was.

Reply:
I've asked Lexi about this too

Reply:
Maybe, it's a tricky one though as it just won't work with some of our test users otherwise. So, guess it has to go in really.

Reply:
https://github.com/hmrc/emcs/blob/main/emcs-backend/src/main/java/uk/gov/hmrc/portal/emcs/business/service/validation/trader/TraderValidationRule.java prefixes used from here